zebra/zebra_srv6.c

```diff
@@ -204,7 +232,7 @@ void zebra_notify_srv6_locator_delete(struct srv6_locator *locator)
 	 * by ZEBRA_SRV6_LOCATOR_DELETE, and this notification is sent to the
 	 * owner of each chunk.
 	 */
-	for (ALL_LIST_ELEMENTS_RO((struct list *)locator->chunks, n, c)) {
+	for (ALL_LIST_ELEMENTS_RO((struct list *)locator->locator.chunks, n, c)) {
```

Comment:
Agree.

I fixed this by adding the required fields to the existing `srv6_locator` data structure instead of creating a nested data structure.

Looks better now.